cpp/include/Ice/SSL/ConnectionInfo.h

```diff
@@ -19,43 +19,127 @@
 
 namespace Ice::SSL
 {
+#if defined(ICE_USE_SCHANNEL)
     /**
      * Provides access to the connection details of an SSL connection.
      */
-    class ICE_API ConnectionInfo : public Ice::ConnectionInfo
+    class ICE_API SchannelConnectionInfo : public Ice::ConnectionInfo
     {
     public:
-        ConnectionInfo() = default;
+        SchannelConnectionInfo() = default;
 
         /**
          * One-shot constructor to initialize all data members.
          * @param underlying The information of the underlying transport or null if there's no underlying transport.
          * @param incoming Whether or not the connection is an incoming or outgoing connection.
          * @param adapterName The name of the adapter associated with the connection.
          * @param connectionId The connection id.
-         * @param certs The certificate chain.
+         * @param peerCertificate The peer certificate.
          */
-        ConnectionInfo(
+        SchannelConnectionInfo(
             const Ice::ConnectionInfoPtr& underlying,
             bool incoming,
             const std::string& adapterName,
             const std::string& connectionId,
-            const std::vector<CertificatePtr>& certs)
+            PCCERT_CONTEXT peerCertificate)
             : Ice::ConnectionInfo(underlying, incoming, adapterName, connectionId),
-              certs(certs)
+              peerCertificate(peerCertificate)
         {
         }
 
-        ~ConnectionInfo() override;
+        ~SchannelConnectionInfo() override;
 
-        ConnectionInfo(const ConnectionInfo&) = delete;
-        ConnectionInfo& operator=(const ConnectionInfo&) = delete;
+        SchannelConnectionInfo(const SchannelConnectionInfo&) = delete;
+        SchannelConnectionInfo& operator=(const SchannelConnectionInfo&) = delete;
 
         /**
-         * The certificate chain.
+         * The peer certificate.
          */
-        std::vector<CertificatePtr> certs;
+        PCCERT_CONTEXT peerCertificate = nullptr;
```

Comment:
Public fields is not a common / recommended style in C++.

It's ok for plain struct but not for such a class. It should be instead:
```cpp
public:
    PCCERT_CONTEXT peerCertificate() const noexcept { return _peerCertificate; }

private:
    const PCCERT_CONTEXT _peerCertificate = nullptr;
```

Reply:
The base class has public fields for other properties, should we fix them in a follow-up PR?